### PR TITLE
Fix metadata

### PR DESCRIPTION
**Pull request by Brean0**
Reverts the Revert BeanstalkFarms/Beanstalk#588 
- fixes metadata to properly display SVG on various NFT marketplaces
- fixed bug to properly support negative stems. 
- added scientific notation for large numbers.

### CODE REVIEW

File: protocol/contracts/beanstalk/metadata/MetadataFacet.sol [hunk 1]
```diff
@@ -5,8 +5,10 @@
 pragma solidity ^0.7.6;
 pragma experimental ABIEncoderV2;
 
-import {LibLegacyTokenSilo} from "contracts/libraries/Silo/LibLegacyTokenSilo.sol";
 import "./MetadataImage.sol";
+import "contracts/mocks/MockERC1155.sol";
```
Comment (publiuss) [inline protocol/contracts/beanstalk/metadata/MetadataFacet.sol:9]: Why is a Mock contract imported in a main facet? Should this be removed?

File: protocol/contracts/beanstalk/metadata/MetadataImage.sol [hunk 2]
```diff
@@ -524,13 +526,93 @@ contract MetadataImage {
...
-                '<text x="230" y="14.5" font-size="12" fill="White" text-anchor="end" font-family="futura">Stem: ',
-                uint256(stem).toString(),
+                '<text x="235" y="14.5" font-size="12" fill="White" text-anchor="end" font-family="futura">Stem: ',
+                SciNotation(stem),
                 '</text>'
             )
         );
     }
 
+    function SciNotation(int96 stem) internal pure returns (string memory) {
```
Comment (publiuss) [inline protocol/contracts/beanstalk/metadata/MetadataImage.sol:536]: Why does this function start with an uppercase character?

File: protocol/contracts/beanstalk/metadata/MetadataImage.sol [hunk 3]
```diff
@@ -524,13 +526,93 @@ contract MetadataImage {
...
+        } else if(stem < 1e26) {
+            return stemDecimals(stem, 25);
+        } else if(stem < 1e27) {
+            return stemDecimals(stem, 26);
+        } else if(stem < 1e28) {
+            return stemDecimals(stem, 27);
+        } else {
+            return stemDecimals(stem, 28);
+        }
+    }
```
Comment (publiuss) [inline protocol/contracts/beanstalk/metadata/MetadataImage.sol:605]: can you a blank line between these functions?

File: protocol/test/Silo.test.js [hunk 4]
```diff
@@ -361,7 +361,7 @@ describe('Silo', function () {
 
     it("properly gives an URI", async function () {
       await this.season.farmSunrises(1000); 
-
+
```
Comment (publiuss) [inline protocol/test/Silo.test.js:364]: Why was a tab added here?

File: protocol/contracts/libraries/LibStrings.sol [hunk 5]
```diff
@@ -51,4 +51,15 @@ library LibStrings {
     function toHexString(address addr) internal pure returns (string memory) {
         return toHexString(uint256(uint160(addr)), _ADDRESS_LENGTH);
     }
+
+    /**
+     * @dev Converts a `int256` to its ASCII `string` representation.
+     */
+    function toString(int256 value) internal pure returns(string memory){
+        if(value > 0){
```
Comment (publiuss) [inline protocol/contracts/libraries/LibStrings.sol:59]: Is there a reason you don't put spaces before and after the parentheses in your if statements?

File: protocol/contracts/beanstalk/metadata/MetadataFacet.sol [hunk 6]
```diff
@@ -19,6 +21,10 @@ import "./MetadataImage.sol";
  */
 contract MetadataFacet is MetadataImage {
     using LibStrings for uint256;
+    using LibStrings for int256;
+
+
```
Comment (publiuss) [inline protocol/contracts/beanstalk/metadata/MetadataFacet.sol:26]: Can you remove one of these blank lines?

File: protocol/contracts/beanstalk/metadata/MetadataFacet.sol [hunk 7]
```diff
@@ -27,26 +33,35 @@ contract MetadataFacet is MetadataImage {
      * Deposits are stored as a mapping of a uint256 to a Deposit struct.
      * ERC20 deposits are represented by the concatination of the token address and the stem. (20 + 12 bytes).
      */
-    function uri(uint256 depositId) external view returns (string memory) {
+    function uri(uint256 depositId) public view returns (string memory) {
```
Comment (publiuss) [inline protocol/contracts/beanstalk/metadata/MetadataFacet.sol:36]: Why was this make `public`? It doesn't seem to be used anywhere else.

Comment (Brean0) [reply protocol/contracts/beanstalk/metadata/MetadataFacet.sol]: fixed: 4e04afe74843ddd502168d255a4fd326e49b41ec

File: protocol/contracts/beanstalk/metadata/MetadataFacet.sol (hunk 6 above)
Comment (Brean0) [reply protocol/contracts/beanstalk/metadata/MetadataFacet.sol]: fixed: c58be9b7fa3eda06805602e40e1e93f0fb281137

File: protocol/contracts/libraries/LibStrings.sol (hunk 5 above)
Comment (Brean0) [reply protocol/contracts/libraries/LibStrings.sol:59]: mistake, fixed to match style guide: c58be9b7fa3eda06805602e40e1e93f0fb281137

File: protocol/test/Silo.test.js (hunk 4 above)
Comment (Brean0) [reply protocol/test/Silo.test.js]: fixed:  1e026b24dfec3afe727b16703a41f9829d6b23ff

File: protocol/contracts/beanstalk/metadata/MetadataImage.sol (hunk 3 above)
Comment (Brean0) [reply protocol/contracts/beanstalk/metadata/MetadataImage.sol:605]: fixed: c58be9b7fa3eda06805602e40e1e93f0fb281137

File: protocol/contracts/beanstalk/metadata/MetadataImage.sol (hunk 2 above)
Comment (Brean0) [reply protocol/contracts/beanstalk/metadata/MetadataImage.sol]: mistake, fixed: c58be9b7fa3eda06805602e40e1e93f0fb281137

File: protocol/contracts/beanstalk/metadata/MetadataFacet.sol (hunk 1 above)
Comment (Brean0) [reply protocol/contracts/beanstalk/metadata/MetadataFacet.sol]: left over from testing, fixed: 9440bbb5c92bdd59b91e39e950699e3a0baaa4f8

**Pull request merged**